app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,7 @@
+class RecipesController < ApplicationController
+  def show
+    @recipe = Recipe.find(params[:id])
+    @ingredients = @recipe.ingredients
```

Comment:
How you have these instance variables set up is totally fine, but just also want to clarify that's it's totally okay to call `@recipe.ingredients` in the view. Doing something like `@recipe.ingredients.order(:name)` in the view would be adding an AR query so that's not cool, but just calling an association or model method in the view is totally fine.